pkg/agent/controller/egress/egress_controller_test.go

```diff
@@ -269,18 +277,24 @@ func TestSyncEgress(t *testing.T) {
 				mockOFClient.EXPECT().InstallPodSNATFlows(uint32(1), net.ParseIP(fakeLocalEgressIP1), uint32(1))
 				mockOFClient.EXPECT().InstallPodSNATFlows(uint32(2), net.ParseIP(fakeLocalEgressIP1), uint32(1))
 				mockRouteClient.EXPECT().AddSNATRule(net.ParseIP(fakeLocalEgressIP1), uint32(1))
-				mockIPAssigner.EXPECT().UnassignIP(fakeLocalEgressIP1)
+				mockIPAssigner.EXPECT().UnassignIP(fakeLocalEgressIP1).Return(true, nil)
 
 				mockOFClient.EXPECT().UninstallSNATMarkFlows(uint32(1))
 				mockRouteClient.EXPECT().DeleteSNATRule(uint32(1))
 				mockOFClient.EXPECT().UninstallPodSNATFlows(uint32(1))
 				mockOFClient.EXPECT().UninstallPodSNATFlows(uint32(2))
-				mockIPAssigner.EXPECT().UnassignIP(fakeLocalEgressIP1)
+				mockIPAssigner.EXPECT().UnassignIP(fakeLocalEgressIP1).Return(true, nil)
 				mockOFClient.EXPECT().UninstallEgressQoS(uint32(1))
 
 				mockOFClient.EXPECT().InstallPodSNATFlows(uint32(1), net.ParseIP(fakeLocalEgressIP1), uint32(0))
 				mockOFClient.EXPECT().InstallPodSNATFlows(uint32(3), net.ParseIP(fakeLocalEgressIP1), uint32(0))
-				mockIPAssigner.EXPECT().UnassignIP(fakeLocalEgressIP1)
+				mockIPAssigner.EXPECT().UnassignIP(fakeLocalEgressIP1).Return(true, nil)
+			},
+			egressEvents: []expectedEvents{
+				{
+					expectedMessage: "Unassigned Egress egressA with IP 1.1.1.1 from Node node1",
+					messageCount:    3,
+				},
```

Comment:
The expectation is wrong. The Egress is a static one, not assigned from a pool, `UnassignIP` should return `false` and no events should be received.

You should only make the Egresses whose IPs are got from IPPool generate events. And the test procedure synced the Egress 3 times, the last of which is a dummy operation to ensure no operation will be executed, typically the last Unassign should also return false.

Please revisit all test cases to ensure we get expected, non-duplicate events.